dart/utils/SkelParser.cpp

```diff
@@ -2293,6 +2301,94 @@ JointPropPtr readTranslationalJoint(
   return dynamics::TranslationalJoint::Properties::createShared(properties);
 }
 
+//==============================================================================
+JointPropPtr readTranslationalJoint2d(
+    tinyxml2::XMLElement* _jointElement,
+    SkelJoint& _joint,
+    const std::string& _name)
+{
+  assert(_jointElement != nullptr);
+
+  dynamics::TranslationalJoint2d::Properties properties;
+
+  //--------------------------------------------------------------------------
+  // Plane
+  if (hasElement(_jointElement, "plane"))
+  {
+    tinyxml2::XMLElement* planeElement = getElement(_jointElement, "plane");
+
+    // Type attribute
+    std::string type = getAttributeString(planeElement, "type");
+
+    if (type == "xy")
+    {
+      properties.mPlaneType = dynamics::PlanarJoint::PlaneType::XY;
```

Comment:
If we do make the member variables private, we'll need to tweak the implementation of this parser to store these parameters and then pass them into the member functions of `properties`.